### PR TITLE
Introduce AccelBLAS template

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Introduce AccelBLAS template

## What type(s) of changes does this code introduce?
- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
@PDoakORNL review ready.

**Comment by ye-luo** [comment]
Test this please

File: src/Platforms/CUDA/AccelBLAS_CUDA.hpp [hunk 1]
```diff
@@ -9,36 +9,38 @@
...
+namespace compute
+{
+template<>
+class BLASHandle<PlatformKind::CUDA>
 {
-  // CUDA specific variables
-  compute::Queue<PlatformKind::CUDA> queue;
+  public:
+  // cuda stream, not owned, reference-only
+  cudaStream_t h_stream;
```
Comment (PDoakORNL) [inline src/Platforms/CUDA/AccelBLAS_CUDA.hpp:29]: the type is actually a pointer, the semantics are very handle like.  From an English standpoint your might say it refers to a stream but in C++ context I wouldn't really call it a reference.

**Review by PDoakORNL — APPROVED** [review]
Looks good to me.

File: src/Platforms/CUDA/AccelBLAS_CUDA.hpp (hunk 1 above)
Comment (ye-luo) [reply src/Platforms/CUDA/AccelBLAS_CUDA.hpp:29]: Not C++ reference type but conceptually a reference.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**